### ISSUE TITLE
CleanJobs() doesn't clean jobs based on age

### ISSUE THREAD

**Issue by michaelrsweet** [apple/cups#1010]
Version: 1.2-current
CUPS.org User: pfournier.loups

Bug still seems present in CVS for 1.1 and 1.2.

We use the standard MaxJobs of 500 and most new jobs are deleted soon after being printed while older jobs dating back to September are kept.

The problem appeared when we started to use varying jobs priority. lower priority jobs are thus put at the end of the linked list used by CleanJobs() and are cleaned out only after all high priority done jobs have already been deleted. We thus can almost never reprint an high priority job from Cups.
